core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -158,6 +170,7 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         stageIds = jobStart.stageIds,
         jobGroup = jobGroup,
         status = JobExecutionStatus.RUNNING)
+    jobGroupToJobIds.getOrElseUpdate(jobGroup.orNull, new HashSet[JobId]).add(jobStart.jobId)
```

Comment:
The `null` job group contains jobs that do not have any other job group set (e.g. jobs without a job group).